src/systems/userinput/devices/keyboard.js

```diff
@@ -45,7 +55,10 @@ export class KeyboardDevice {
         this.keys = {};
         this.seenKeys.clear();
       } else {
-        const key = event.key.toLowerCase();
+        let key = event.key.toLowerCase();
+        // Use event.code and not event.key for wasdqe controls so this works
+        // for non QWERTY keyboards.
+        key = (event.code && CODE_TO_KEY[event.code]) || key;
```

Comment:
```suggestion
        // Use event.code and not event.key for wasdqe controls so this works
        // for non QWERTY keyboards.
        const key = CODE_TO_KEY[event.code] ?? event.key.toLowerCase();
```
`event.code` always truthy in modern browser, so `key = CODE_TO_KEY[event.code] || event.key.toLowerCase();`
`event.code` always falsy in old browser, so `key = event.key.toLowerCase();`

You can support backward compatibility with:
```
export const KEYCODE_TO_KEY = {
  65: "q",
  68: "d",
  69: "e",
  81: "a",
  83: "s",
  90: "w"
};
// https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code#browser_compatibility
// https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode#browser_compatibility
const key = CODE_TO_KEY[event.code] ?? KEYCODE_TO_KEY[event.keycode] ?? event.key.toLowerCase()
```

Reply:
You are mixing two different things, code "KeyW", and keycode "90". There is no need for a mapping between keycode to key here, we already have to key in `event.key`.

"event.code always truthy in modern browser, so key = CODE_TO_KEY[event.code] || event.key.toLowerCase();
event.code always falsy in old browser, so key = event.key.toLowerCase();"
Yes this is exactly what I want. For old browsers, they don't have the required thing to know which key is pressed (event.code), then we default to the wrong qwerty behavior, no choice.

Reply:
For your other way to write it
`const key = CODE_TO_KEY[event.code] ?? event.key.toLowerCase();`
This should work I guess yes.
I'll leave this decision to Mozilla if they decide to include a fix for this issue. I won't spend time to modify this PR.

Reply:
I've test it works and more clean for the final JS code. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

Yes you are right for the backward thing.